src/neon/stage1.rs

```diff
@@ -0,0 +1,531 @@
+#![allow(dead_code)]
+
+use crate::neon::intrinsics::*;
+use crate::neon::utf8check::*;
+use crate::*;
+use std::mem;
+
+pub const SIMDJSON_PADDING: usize = mem::size_of::<uint8x16_t>() * 4;
+
+#[derive(Debug)]
+struct SimdInput {
+    v0: uint8x16_t,
+    v1: uint8x16_t,
+    v2: uint8x16_t,
+    v3: uint8x16_t,
+}
+
+fn fill_input(ptr: &[u8]) -> SimdInput {
+    unsafe {
+        #[allow(clippy::cast_ptr_alignment)]
+            SimdInput {
+            v0: vld1q_u8(ptr.as_ptr() as *const u8),
+            v1: vld1q_u8(ptr.as_ptr().add(16) as *const u8),
+            v2: vld1q_u8(ptr.as_ptr().add(32) as *const u8),
+            v3: vld1q_u8(ptr.as_ptr().add(48) as *const u8),
+        }
+    }
+}
+
+#[cfg_attr(not(feature = "no-inline"), inline(always))]
+unsafe fn neon_movemask(input: uint8x16_t) -> u16 {
+    let bit_mask: uint8x16_t = uint8x16_t::new(0x01, 0x02, 0x4, 0x8, 0x10, 0x20, 0x40, 0x80,
```

Comment:
We have this mask in a few places, perhaps we can `const` it out and use a load instruction to load it in a register? (I see it twice in this file and once in `deser.rs` I think)